src/vos/vos_io.c

```diff
@@ -1464,7 +1453,6 @@ vos_update_end(daos_handle_t ioh, uint32_t pm_ver, daos_key_t *dkey, int err,
 	/* Publish SCM reservations */
 	if (ioc->ic_actv_at != 0) {
 		err = umem_tx_publish(umem, ioc->ic_actv, ioc->ic_actv_at);
-		ioc->ic_actv_at = 0;
```

Comment:
this line was causing the following to happen

evtree was detecting that an invalid overwrite and returning an error
umem_cancel was calling umem_free because this was 0.

pmdk_debug threw an assertion.  I imagine we were seeing random issues otherwise.